t/03-auth-openid.t

```diff
@@ -25,12 +26,36 @@ $c->set_always(session => \%session);
 ok OpenQA::WebAPI::Auth::OpenID::_handle_verified($c, $vident), 'can call _handle_verified';
 $users->called_ok('create_user', 'new user is created for initial login');
 is(($users->call_args(2))[1], 'mordred', 'new user created with details');
+
 $c->set_always(
     req => Test::MockObject->new->set_always(params => Test::MockObject->new->set_always(pairs => [1, 2]))
       ->set_always(url => Test::MockObject->new->set_always(base => 'openqa')))
   ->set_always(app => Test::MockObject->new->set_always(config => {})
       ->set_always(log => Test::MockObject->new->set_true('error', 'debug')))->set_true('flash');
-is OpenQA::WebAPI::Auth::OpenID::auth_response($c), 0, 'can call auth_response';
-$c->app->log->called_ok('error', 'an error was logged for call without proper config');
+stderr_like {
+    my $ret = OpenQA::WebAPI::Auth::OpenID::auth_response($c);
+    is($ret, 0, 'can handle not_openid response on invalid OpenID message');
+}
+qr/Invalid OpenID/, 'warning about invalid OpenID is displayed';
+
+$c->set_always(
+    req => Test::MockObject->new->set_always(params => Test::MockObject->new->set_always(pairs => [1, 2]))
+      ->set_always(url => Test::MockObject->new->set_always(base => 'openqa'))
+      ->set_always(log => Test::MockObject->new->set_true('error', 'debug'))->set_true('flash'));
```

Comment:
this looks like mostly duplicated from lines L28-32. Better reuse the original mocked controller and only overwrite the function you need.

Reply:
done. 11e55fac4